bin/version-include.sh

```diff
@@ -0,0 +1,104 @@
+# Copyright © 2021, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+function populateValuesYAML() {
+  v4mValuesYAML=$1
+  rm -f "$v4mValuesYAML"
+  touch "$v4mValuesYAML"
+
+  # Attempt to obtain current git commit hash
+  gitCommit=$(git rev-parse --short HEAD 2>/dev/null)
+  if [ -n "$gitCommit" ]; then
+    echo "gitCommit: $gitCommit" >> "$v4mValuesYAML"
+  fi
+
+  # List contents of USER_DIR
+  if [ -d "$USER_DIR" ]; then
+    echo '"user_dir":' >> "$v4mValuesYAML"
+    echo "  path: $(dirname $USER_DIR)" >> "$v4mValuesYAML"
+    echo '  files: |' >> "$v4mValuesYAML"
+    l=($(find "$USER_DIR" -type f))
+    for (( i=0; i<${#l[@]}; i++ )); do
+      fullPath=${l[i]}
+      path=${fullPath#"$USER_DIR/"}
+      echo "      $path" >> "$v4mValuesYAML"
+    done
+  fi  
+  # Top-level user.env contents
+  if [ -f "$USER_DIR/user.env" ]; then
+    echo '  "user.env": |' >> "$v4mValuesYAML"
+    cat "$USER_DIR/user.env" | sed 's/^/      /' >> "$v4mValuesYAML"
+  fi
+  # Monitoring user.env contents
+  if [ -f "$USER_DIR/monitoring/user.env" ]; then
+    echo '  "monitoring_user.env": |' >> "$v4mValuesYAML"
+    cat "$USER_DIR/monitoring/user.env" | sed 's/^/      /' >> "$v4mValuesYAML"
+  fi
+  # Logging user.env contents
+  if [ -f "$USER_DIR/logging/user.env" ]; then
+    echo '  "logging_user.env": |' >> "$v4mValuesYAML"
+    cat "$USER_DIR/logging/user.env" | sed 's/^/      /' >> "$v4mValuesYAML"
+  fi
+}
+
+function deployV4MInfo() {
+  NS=$1
+  if [ -z "$NS" ]; then
+    log_error "No namespace specified for deploying version info"
+    return 1
+  fi
+
+  valuesYAML=$TMP_DIR/v4mValues.yaml
+  populateValuesYAML "$valuesYAML"
+
+  log_info "Updating version info..."
+  helm upgrade --install \
+    -n "$NS" \
+    --values $valuesYAML \
+    v4m ./v4m-chart
+}
+
+function removeV4MInfo() {
+  NS=$1
+  if [ -z "$NS" ]; then
+    log_error "No namespace specified for removing version info"
+    return 1
+  fi
+  log_info "Removing version info..."
+  helm uninstall -n "$NS" v4m
+}
+
+if [ -z "$V4M_VERSION_INCLUDE" ]; then
+  origIFS=$IFS
+  IFS=$'\n' v4mHelmVersionLines=($(helm list -n "$V4M_NS" --filter '^v4m$' -o yaml))
+  IFS=$origIFS
+  if [ -z "$v4mHelmVersionLines" ]; then
+  log_debug "No v4m release found in [$V4M_NS]"
+  else
+    for (( i=0; i<${#v4mHelmVersionLines[@]}; i++ )); do 
+      line=${v4mHelmVersionLines[$i]}
+      vre='app_version: (([0-9]+).([[0-9]+).([0-9]+)\.?(-.+)?)'
+      sre='status: (.+)'
+      if [[ $line =~ $vre ]]; then
+        V4M_CURRENT_VERSION_FULL=${BASH_REMATCH[1]}
+        V4M_CURRENT_VERSION_MAJOR=${BASH_REMATCH[2]}
+        V4M_CURRENT_VERSION_MINOR=${BASH_REMATCH[3]}        
+        V4M_CURRENT_VERSION_PATCH=${BASH_REMATCH[4]}
+      elif [[ "$line" =~ $sre ]]; then
+        V4M_CURRENT_STATUS=${BASH_REMATCH[1]}
+      fi
+    done
+
+    log_debug "V4M_CURRENT_VERSION_FULL=$V4M_CURRENT_VERSION_FULL"
```

Comment:
If there is no previous release found, do we still want all of these debug messages with blank values?

Reply:
The if block above handles the case where there is no existing release and this info will not get output.